### ISSUE TITLE
Generalize subcomponent() and neighborhood()

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1854]
**What is the feature or improvement you would like to see?**

`igraph_subcomponent()` finds the list of vertices which are reachable from a _single_ vertex. `igraph_neighborhood()` finds the list of vertices which are reachable from a vertex within a given (non-infinite) distance.

There is an obvious useful generalization for both:

 1. Allow more than one vertex as the starting point. Which vertices are reachable from any of them?
 2. Allow infinity as the distance. In this case, `neighborhood()` becomes equivalent to `subcomponent()`.

Now we have two functions, but we could in principle combine the functionality of both into a single function _and_ generalize it at the same time (allow more than one starting vertex).

This issue is to discuss if we want to do this for 0.10, as it would involve breaking the API.

We might want to avoid an interface that is too overloaded and perhaps generalize only `subcomponent()` to support: (1) more than one starting vertex (2) a maximum cutoff distance. But we would not include the `mindist` feature of `neighbourhood()`, or support starting from more than one set of vertices at the same time (as `neighborhood()` does).

Obviously, I am getting inspired by Mathematica's `Vertex(In|Out)Component[]`, whose documentation you might take a look at: http://reference.wolfram.com/language/ref/VertexOutComponent.html

The main feature I am looking for here is the capability to be able to find vertices reachable from an entire set of vertices (as opposed to a single vertex). Currently, this can only be done with many calls to `subcomponent()` and combining the result. This is both cumbersome and slow (worse performance than what is possible). Or am I missing an existing function?

**Comment by ntamas** [comment]
> am I missing an existing function?

I don't think so. There's `igraph_bfs`, which you can call with a set of vertices to start the BFS from, but note that these vertices will be considered one by one (i.e. igraph starts the traversal from the second vertex when it is done with the first one), so it's not the same.
